src/lib.rs

```diff
@@ -1696,6 +1696,86 @@ pub fn strongly_connected_components(
         .collect()
 }
 
+/// Return the first cycle encountered during DFS of a given PyDiGraph
+///
```

Comment:
We probably should add a note here that the list will be empty if there is no cycle found. I think networkx raises an exception which we could do too. It doesn't really matter which way we go about it, just that we document the behavior.